### PR TITLE
HDDS-11457. Internal error on S3 CompleteMultipartUpload if parts are not specified

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

When submitting `CompleteMultipartUpload` request without parts, `CompleteMultipartUploadRequestUnmarshaller` encounters `SAXParseException`, which it converts to error response with the default `500 Internal Server Error` code.  We can use the more specific `InvalidRequest` code instead.  Similar problem is fixed in two other unmarshallers.  Additionally, we can avoid the `SAXParseException` by checking for empty input and returning more specific error message (`You must specify at least one part`) for empty `CompleteMultipartUpload` request.

https://issues.apache.org/jira/browse/HDDS-11457

## How was this patch tested?

Tweaked existing robot test to verify this case.

CI:
https://github.com/adoroszlai/ozone/actions/runs/10846339715

### CODE REVIEW

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java [hunk 1]
```diff
@@ -116,4 +118,11 @@ public static S3StorageType toS3StorageType(String storageType)
       throw newError(INVALID_ARGUMENT, storageType, ex);
     }
   }
+
+  public static WebApplicationException wrapOS3Exception(OS3Exception ex) {
+    return new WebApplicationException(ex.getErrorMessage(), ex,
+        Response.status(ex.getHttpCode())
+            .entity(ex.toXml())
+            .build());
+  }
```
Comment (ivandika3) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java:127]: Nit: `AuthorizationFilter` also seems to use the same method, we can also replace the `wrapOS3Exception` in `AuthorizationFilter` with this.

**Review by ivandika3 — COMMENTED** [review]
@adoroszlai Thank you for the patch. Overall LGTM.

**Review by ivandika3 — APPROVED** [review]
Thanks for the update. LGTM +1.

**Comment by adoroszlai** [comment]
Thanks @ivandika3 for the review.

**Pull request merged**